File: JCleanArchitectureApp/appkoin/src/main/java/com/woowrale/jcleanarchitecture/data/remote/server/GetRemoteContactSource.java
package com.woowrale.jcleanarchitecture.data.remote.server;

import com.woowrale.data.repository.remote.RemoteContactSource;
import com.woowrale.domain.model.Contact;
import com.woowrale.jcleanarchitecture.data.remote.mappers.MapperToContacts;
import com.woowrale.jcleanarchitecture.data.remote.ws.ApiService;

import java.io.IOException;
import java.util.List;

public class GetRemoteContactSource implements RemoteContactSource {

    private ApiService apiService;

    public GetRemoteContactSource(ApiService apiService) {
        this.apiService = apiService;
    }

    @Override
    public List<Contact> getContacts(String apiContacts, String source, String query) {
        try {
            return new MapperToContacts().mappertoContactList(apiService.getContacts(apiContacts, source, query).execute().body());
        } catch (IOException e) {
            return null;
        }
    }
}

File: KCleanArchitectureApp/appdagger/src/main/java/com/woowrale/kcleanarchitecture/data/remote/datasource/GetRemoteContactSource.kt
package com.woowrale.kcleanarchitecture.data.remote.datasource

import com.woowrale.data.repository.remote.RemoteContactSource
import com.woowrale.domain.model.Contact
import com.woowrale.kcleanarchitecture.BuildConfig
import com.woowrale.kcleanarchitecture.data.remote.mappers.toContact
import com.woowrale.kcleanarchitecture.data.remote.server.ApiService
import java.io.IOException

class GetRemoteContactSource(apiService: ApiService) : RemoteContactSource {

    private val apiService: ApiService = apiService

    override fun getContacts(
        source: String,
        query: String
    ): List<Contact> {
        return try {
            apiService.getContacts(BuildConfig.GET_CONTACTS, source).execute().body()!!.map{ it.toContact() }
        } catch (e: IOException) {
           return listOfNotNull()
        }
    }

}

